Import a molecule to the server by an identifier

psavery opened the pull request:
A molecule can be imported by its inchikey, inchi,
or smiles.

Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

psavery committed:
Import a molecule from the server by an identifier

A molecule can be imported by its inchikey, inchi,
or smiles.

Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

psavery commented:
@cjh1 I haven't checked anything about the molecule, yet, but I was able to import a molecule like this:

```
import openchemistry as oc
  
# Search for ethanol. Make sure it is in the database first.
smiles = 'CCO'

molecule = oc.import_structure(smiles=smiles)

print('molecule is', molecule)
# molecule is <openchemistry.GirderMolecule object at 0x7fc6c3829dd8>
```

psavery commented:
Also, importing by SMILES only works with [this PR](https://github.com/OpenChemistry/mongochemserver/pull/102) on mongochemserver.

cjh1 commented on openchemistry/__init__.py:
```diff
@@ -668,6 +668,32 @@ def _find_using_cactus(identifier):
+        raise Exception('At least one of these must be set: '
+                        'inchikey, inchi, or smiles')
+
+    molecules = girder_client.get('molecules', parameters=params)
```
Can we use the POST ( create endpoint? ) here?

psavery replied:
Oh yes... I have that in my notes. Sorry I missed that.

psavery replied:
It is fixed. Now, if the molecule doesn't exist in the database, one will be created.

psavery committed:
Import molecules via POST instead of GET

With this change, if the molecule is not found in the database,
one will be created.

The CJson is also obtained from the POST.

Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

psavery committed:
Remove inchikey from import_structure() parameters

import_structure() is meant to import a structure
into the database via a POST, so inchikey should
not be a parameter, since it would be for a GET.

Signed-off-by: Patrick Avery <patrick.avery@kitware.com>

cjh1 commented:
@psavery As a follow up PR lets add `smiles` and `inchikey` to the parameter list for `find_structure`

psavery commented:
Okay, will do!
